File: frontoffice/src/app/features/engagement/services/event.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { EventDTO, EventRegistration } from '../models/engagement.models';

const BASE = '/api';

@Injectable({ providedIn: 'root' })
export class EventService {
  constructor(private http: HttpClient) {}

  getAllEvents(): Observable<EventDTO[]> {
    return this.http.get<EventDTO[]>(`${BASE}/events`);
  }

  getEventsByGovernorate(gov: string): Observable<EventDTO[]> {
    return this.http.get<EventDTO[]>(`${BASE}/events/governorate/${gov}`);
  }

  createEvent(dto: EventDTO): Observable<EventDTO> {
    return this.http.post<EventDTO>(`${BASE}/events`, dto);
  }

  registerToEvent(eventId: number): Observable<EventRegistration> {
    return this.http.post<EventRegistration>(`${BASE}/events/${eventId}/register`, {});
  }

  markAttended(regId: number): Observable<void> {
    return this.http.patch<void>(`${BASE}/events/registrations/${regId}/attend`, {});
  }

  getMyRegistrations(): Observable<number[]> {
    return this.http.get<number[]>(`${BASE}/events/my-registrations`);
  }
}

File: Backend Trusted/module06-engagement/src/main/java/com/trustedwork/module06/service/EventService.java
package com.trustedwork.module06.service;

import com.trustedwork.module06.dto.EventDTO;
import com.trustedwork.module06.entity.EventRegistration;
import java.util.List;

public interface EventService {
    EventDTO createEvent(EventDTO dto);
    List<EventDTO> getAllEvents();
    List<EventDTO> getEventsByGovernorate(String governorate);
    EventRegistration registerToEvent(Long eventId, Long userId);
    void markAttended(Long registrationId, Long userId);
    EventDTO updateEvent(Long id, EventDTO dto);
    void deleteEvent(Long id);
    List<Long> getMyRegisteredEventIds(Long userId);
    void cancelRegistration(Long eventId, Long userId);
}
